File: Timesheet/ClientApp/src/models/Activity.ts
import { Stringified } from "../utilities/Strings";

export enum ActivityId {
    Plan = 1,
    Design,
    Development,
    Test
}

export type Activity = {
    id: ActivityId;
    name: string;
};

export function stringifyActivity(activity : Activity) : Stringified<Activity> {
    return {
        ...activity,
        id : activity.id.toString(),
    }
}

File: Timesheet.Data/Models/Activity.cs
using System.Collections.Generic;

namespace Timesheet.Data.Models
{
    public enum ActivityId
    {
        Plan = 1,
        Design,
        Development,
        Test
    }

    public class Activity
    {
        public const int NameMaxLength = 200;

        public ActivityId Id { get; set; }

        public string Name { get; set; } = null!;

        public virtual ICollection<WorkLog> WorkLogs { get; } = null!;
    }
}